src/UpcomingEvents.js

```diff
@@ -12,13 +12,20 @@ function UpcomingEvents() {
 
   return (
     <div className="event-card__wrapper">
-      {upcomingEvents.map(function(obj, i) {
-        return (
-          <a href={`/events/${obj.path}`} className="event-card__wrapper--link">
-            <EventCard key={i} row={obj} />
-          </a>
-        );
-      })}
+      {upcomingEvents.length === 0 ? (
```

Comment:
This is causing a flash that displays "There are no upcoming events."

Reply:
https://github.com/HubSpot/cms-event-registration/blob/bc0cf959f724d783f8a45fbd1eaad17e9cecb77c/src/components/RegisteredEventListings.js#L32 is one way we've handled this in the past

Reply:
Yeah, I think we need to keep a status in the state. In general, it is best for it be a an enum like `RequestStatus` with states like `UNINITIALIZED`, `FETCHING`, `SUCCEEDED`, `FAILED`.

Reply:
@gcorne, I just implemented a Loading indicator based on @anthmatic's suggestion. I'm happy to refactor this to include an enum like `Request Status`, but I think I'd need advice on how to implement this with our current setup of `AppContext.js`. Any help would be much appreciated!